.github/workflows/java.yaml

```diff
@@ -14,23 +14,27 @@ jobs:
   build:
     name: Build
     runs-on: ${{ matrix.os }}
+    container: ${{ matrix.container }}
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest]
+        include:
+          - os: ubuntu-latest
+            # Use Fedora container to get new enough Meson
+            container: registry.fedoraproject.org/fedora:latest
+          - os: macos-latest
     steps:
     - name: Check out repo
       uses: actions/checkout@v3
     - name: Install dependencies (Linux)
       if: ${{ matrix.os == 'ubuntu-latest' }}
-      run: sudo apt-get install libopenslide0-dev
+      run: sudo dnf install -y gcc java-latest-openjdk-devel meson openslide-devel
```

Comment:
So you're installing rpms on ubuntu. Does that actually work right and build things with the right dependencies, or will it now depend on whatever openjdk release fedora uses?
I did notice the ubuntu-latest build check hasn't completed yet while everything else took under a minute.

Reply:
It's not actually Ubuntu, it's a Fedora container running in an Ubuntu host.  But I think you're right that `pip install meson` in the host would be better.